src/BuiltInTools/dotnet-watch/HotReload/CompilationHandler.cs

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Locator;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.ExternalAccess.DotNetWatch;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace Microsoft.DotNet.Watcher.Tools
+{
+    internal class CompilationHandler : IDisposable
+    {
+        private readonly IReporter _reporter;
+        private Task<(Solution, DotNetWatchEditAndContinueWorkspaceService)>? _initializeTask;
+        private Solution? _currentSolution;
+        private DotNetWatchEditAndContinueWorkspaceService? _editAndContinue;
+        private IDeltaApplier? _deltaApplier;
+
+        public CompilationHandler(IReporter reporter)
+        {
+            _reporter = reporter;
+        }
+
+        public async ValueTask InitializeAsync(DotNetWatchContext context, CancellationToken cancellationToken)
+        {
+            if (_deltaApplier is null)
+            {
+                _deltaApplier = context.DefaultLaunchSettingsProfile.HotReloadProfile == "blazorwasm" ?
+                    new BlazorWebAssemblyDeltaApplier(_reporter) :
+                    new AspNetCoreDeltaApplier(_reporter);
+            }
+
+            await _deltaApplier.InitializeAsync(context, cancellationToken);
+
+            if (context.Iteration == 0)
+            {
+                var instance = MSBuildLocator.QueryVisualStudioInstances().First();
```

Comment:
Is instance equivalent to window here? AKA we can only connect to one?

Reply:
It's basically saying use this version of MSBuild when running builds. Kinda equivalent to what a global.json does.